pkg/utils/request.go

```diff
@@ -31,3 +31,25 @@ func BindJSONAndValidate(ctx *gin.Context, i interface{}) bool {
 
 	return false
 }
+
+func BindURIAndValidate(ctx *gin.Context, i interface{}) bool {
+	err := ctx.ShouldBindUri(i)
+	if err == nil {
+		return true
+	}
+
+	if errValidations, ok := err.(validator.ValidationErrors); ok {
+		res := validations.HandleValidationErrors(errValidations)
+		ctx.JSON(res.Status, res)
+	} else if _, ok := err.(*json.UnmarshalTypeError); ok {
+		ctx.JSON(http.StatusBadRequest, model.WebServiceResponse{
+			Message: "Schema request tidak valid",
+			Status:  http.StatusBadRequest,
+			Data:    nil,
+		})
+	} else {
+		ctx.JSON(http.StatusInternalServerError, ToWebServiceResponse("Internal server error", http.StatusInternalServerError, nil))
+	}
```

Comment:
Ini sebenerny samo be dgn yg bindjson kan, nah minta tolong dipecah jadi function baru be, validate request atau apalaha gitu, biar DRY (don't repeat yourself)